src/main/java/de/danielbechler/diff/selector/CollectionItemElementSelector.java

```diff
@@ -71,6 +94,6 @@ public boolean equals(final Object o)
 	@Override
 	public int hashCode()
 	{
-		return item != null ? item.hashCode() : 0;
+		return item != null ? identityStrategy.hashCode(item) : 0;
```

Comment:
This is the place where things go wrong, when `hashCode` diverges from the contract with `equals`. Since the items hashCode isn't reliably anymore when an `IdentityStrategy` is provided, I recommend changing the hashCode of this selector to a constant. This will certainly have a performance impact on huge collections with lots of changes. But lets make it work before we optimize for performance.